src/redux.js

```diff
@@ -752,8 +807,31 @@ export function getEmptyCellDetails(state) {
 
 export function getTrainedModelDataToSave(state) {
   const dataToSave = {};
+
   dataToSave.name = state.trainedModelDetails.name;
-  dataToSave.description = state.trainedModelDetails.description;
+
+  // If we have column descriptions in metadata, use that, otherwise
+  // use what the user has manually entered.
+  if (state.metadata && state.metadata.fields) {
+    dataToSave.columns = [];
+    for (const columnDescription of getSelectedColumnDescriptions(state)) {
+      dataToSave.columns.push({
+        id: columnDescription.id,
+        description: columnDescription.description
+      });
+    }
+  } else {
+    dataToSave.columns = state.trainedModelDetails.columns;
+  }
+
+  dataToSave.uses = state.trainedModelDetails.uses;
+  dataToSave.misuses = state.trainedModelDetails.misuses;
+
+  dataToSave.identify_subgroup = !!state.trainedModelDetails.identify_subgroup;
```

Comment:
should these be camelCase or like_this? Mostly wondering if they should be consistent and whether I should clean up the fieldsIMade to match these new_ones?

Reply:
Oh, I think camelCase is better; not sure why I did it this way.  Let me update all instances.